validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorStatusLogger.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.api.response.v1.beacon.ValidatorStatus;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.validator.api.ValidatorApiChannel;
+
+public class ValidatorStatusLogger {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final int VALIDATOR_KEYS_PRINT_LIMIT = 20;
+
+  final Set<BLSPublicKey> validatorPublicKeys;
+  final ValidatorApiChannel validatorApiChannel;
+  final AtomicReference<Map<BLSPublicKey, ValidatorStatus>> latestValidatorStatuses =
+      new AtomicReference<>();
+
+  public ValidatorStatusLogger(
+      Set<BLSPublicKey> validatorPublicKeys, ValidatorApiChannel validatorApiChannel) {
+    this.validatorPublicKeys = validatorPublicKeys;
```

Comment:
Probably worth asserting `!validatorPublicKeys.isEmpty()` since we're requiring responses from `validatorApiChannel` below.

Reply:
Done.